### PR TITLE
Fix doxygen check

### PR DESCRIPTION
**Pull request by AniruddhaKanhere**
Description
-----------
This PR updates the doxygen version in use (to 1.9.1).
Also, the config file was updated with `doxygen -u <config file name>`.

Test Steps
-----------

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/doxygen/config.doxyfile [hunk 1]
```diff
@@ -269,7 +269,11 @@ TAB_SIZE               = 4
 # commands \{ and \} for these it is advised to use the version @{ and @} or use
 # a double escape (\\{ and \\})
 
-ALIASES                =
+ALIASES                = "transportcallback=@ingroup mqtt_callback_types" \
+                         "transportstruct=@ingroup mqtt_struct_types" \
+                         "transportpage=@page mqtt_transport_interface Transport Interface" \
+                         "transportsectionimplementation=@section mqtt_transport_interface_implementation Implementing the Transport Interface" \
+                         "transportsectionoverview=@section mqtt_transport_interface_overview Transport Interface Overview"
```
Comment (yanjos-dev) [inline docs/doxygen/config.doxyfile:276]: What are these changes? They seem unrelated to updating the version of doxygen being used.

File: docs/doxygen/config.doxyfile [hunk 2]
```diff
@@ -468,7 +475,7 @@ LOOKUP_CACHE_SIZE      = 0
 # DOT_NUM_THREADS setting.
 # Minimum value: 0, maximum value: 32, default value: 1.
 
-# NUM_PROC_THREADS       = 1
+NUM_PROC_THREADS       = 1
```
Comment (yanjos-dev) [inline docs/doxygen/config.doxyfile:478]: Same as before, is this related to updating the version?

File: docs/doxygen/config.doxyfile (hunk 1 above)
Comment (AniruddhaKanhere) [reply docs/doxygen/config.doxyfile:276]: Since the file was updated automatically, they were copied over from the bottom of the previous file. These are not required. I shall delete them.

File: docs/doxygen/config.doxyfile (hunk 2 above)
Comment (AniruddhaKanhere) [reply docs/doxygen/config.doxyfile:478]: The file was updated automatically by `doxygen -u` command. It made apt changes wherever necessary.

File: docs/doxygen/config.doxyfile (hunk 1 above)
Comment (yanjos-dev) [reply docs/doxygen/config.doxyfile]: Sounds good. I'll approve after the tests are updated.

**Review by yanjos-dev — APPROVED** [review]

**Review by yanjos-dev — APPROVED** [review]

**Review by cobusve — APPROVED** [review]

**Review by archigup — APPROVED** [review]
